Lelik7777/flesh-cards
# Chore/pre push

# Add Pre-Push Hook for Build Verification

## Description of Changes

This pull request introduces a pre-push hook that runs the project build before every push. This ensures that the code compiles without errors before being sent to the remote repository.

## Implementation Details

1. Created the file  '.husky/pre-push' with the following content:

   ``` sh
   #!/usr/bin/env sh
   . "$(dirname -- "$0")/_/husky.sh"

   pnpm run pre-push


2. Added a pre-push script in package.json:

```js
"scripts": {
  "pre-push": "pnpm run lint && pnpm run format && pnpm run build"
} 
````

3. Updated project dependencies to support Husky.

## How It Works
- Before each push, Git triggers the pre-push script.
- The script executes the command pnpm run pre-push.
- This command sequentially runs the linter, formatting, and builds the project.
-  If any of these steps fail, the push is aborted.
## Why This Is Needed
- Prevents the submission of code that fails to compile or does not pass linting checks.
- Improves code quality in the repository.
- Saves time in CI/CD by avoiding known build failures.
## How to Test
- Clone the branch locally.
- Make changes to the code.
- Attempt to push. Ensure that the build and checks are triggered.
- Intentionally introduce an error in the code and confirm that the push is aborted.
## Notes
- Make sure you have the latest version of Husky installed.
- If you need to skip the checks, use the --no-verify flag when pushing.
